packages/wrangler/src/deployments.ts

```diff
@@ -221,26 +231,46 @@ export async function viewDeployment(
 		`/accounts/${accountId}/workers/deployments/by-script/${scriptTag}/detail/${deploymentId}`
 	);
 
-	const flatObj: Record<string, unknown> = {};
-	for (const deployDetailsKey in deploymentDetails) {
-		if (
-			Object.prototype.hasOwnProperty.call(deploymentDetails, deployDetailsKey)
-		) {
-			//@ts-expect-error flattening objects causes the index signature to error
-			const value = deploymentDetails[deployDetailsKey];
-			if (typeof value === "object" && value !== null) {
-				for (const subKey in value) {
-					if (Object.prototype.hasOwnProperty.call(value, subKey)) {
-						flatObj[`${deployDetailsKey}.${subKey}`] = value[subKey];
-					}
-				}
-			} else {
-				flatObj[deployDetailsKey] = value;
-			}
-		}
-	}
-
-	logger.log(flatObj);
+	const triggerStr = deploymentDetails.annotations?.["workers/triggered_by"]
+		? `${formatTrigger(
+				deploymentDetails.annotations["workers/triggered_by"]
+		  )} from ${formatSource(deploymentDetails.metadata.source)}`
+		: `${formatSource(deploymentDetails.metadata.source)}`;
+
+	const rollbackStr = deploymentDetails.annotations
+		? ["workers/rollback_from"]
```

Comment:
I think you meant to do a optional chaining, which can be done with dot notation `?.` however not with bracket notation `objection['bracketNotation']`
```suggestion
	const rollbackStr = deploymentDetails.annotations["workers/rollback_from"]
```